docs/markdown/DevGuide_Data_Memory.md

```diff
@@ -0,0 +1,85 @@
+        ->getRepresentation<BufferRAM>()     // Get RAM representation
+        ->dispatch<std::pair<dvec4, dvec4>>( // Dispatch to produce a pair of vec4s as result
+            [](auto br) { // Deduce actual data type automatically, set ValueType
+                // br's type is deduced to BufferRAMPrecision<ValueType, BufferTarget::Data>
+                // which means it holds values of type ValueType and is a Data buffer
+                // ValueType can be extracted from br's complex type like so:
+                using ValueType = util::PrecisionValueType<decltype(br)>;
+                const auto &data = br->getDataContainer(); // Gets you a std::vector<ValueType>&
+                // Construct result as pair of ValueTypes
+                using Res = std::pair<ValueType, ValueType>;
+                // Initialize result with max and min values of given data type
+                Res minmax{DataFormat<ValueType>::max(),
+                           DataFormat<ValueType>::lowest()};
+                // Iterate over buffer, accumulating the min and max components
+                minmax = std::accumulate(data.begin(), data.end(), minmax,
+                    [](const Res &mm, const ValueType &v) -> Res {
+                        // Take component-wise min and max
+                        return {glm::min(mm.first, v), glm::max(mm.second, v)};
+                });
+                // Convert to pair of dvec4, as we specified in dispatch's type parameter
+                return std::pair<dvec4, dvec4>{util::glm_convert<dvec4>(minmax.first),
+                                               util::glm_convert<dvec4>(minmax.second)};
+        });
+        // minmax now has the lower left and upper right of the meshes bounding box.
+    }
+```
+We start as expected: first the `Mesh` handle is retrieved, then we find the buffer of interest, which is the vertex position buffer in our case. If this buffer exists, we will request its `BufferRAM` representation. Since this representation already exists in an up-to-date state, it can be retrieved without any memory transfer. On this representation the `dispatch<ResultType>(Lambda)` function is called. The actual data access is limited to the Lambda that is passed to `dispatch()` and the result type must match the given type parameter. The actual computations on the data happens inside this Lambda. If any additional variables from the outside scope are required in the Lambda, they can be passed into the Lambdas scope by adding them into the brackets like so:
+```cpp
+rep->dispatch<ResultType>( [ &varFromOutsideScope ] (auto br) { ... })
+```
+The Lambda must also take one parameter, in this case `auto br`, that specifies the actual data type of the elements inside the `Buffer`. By using `auto`, we can let C++ deduce the correct data type and define the operation in a type-agnostic way, so that it works for all different types automatically.
```

Comment:
there is actually no deduction going on here. you just provide a function for each possible type, in this case by giving a generic function that will be compiles for all 40 kind of types. We then just call the correct one depending on the type of the ram representation.